bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java

```diff
@@ -0,0 +1,582 @@
+      dbName = DEFAULT_DB;
+      logger.debug("using default db name {}", DEFAULT_DB);
+    }
+
+    isProperlyConfigured = true;
+
+    connect();
+
+    // check connection; errors will only be logged, hoping the connection will work at a later
+    // time.
+    if (!checkConnection()) {
+      logger.error("database connection does not work for now, will retry to use the database.");
+    }
+  }
+
+  public void deactivate() {
+    logger.debug("influxdb persistence service deactivated");
+    disconnect();
+  }
+
+  private void connect() {
+    if (influxDB == null) {
+      // reuse an existing InfluxDB object because concerning the database it has no state
+      // connection
+      influxDB = InfluxDBFactory.connect(url, user, password);
+      influxDB.enableBatch(200, 100, timeUnit);
+    }
+    connected = true;
+  }
+
+  private boolean checkConnection() {
+    boolean dbStatus = false;
+    if (!connected) {
+      logger.error("checkConnection: database is not connected");
+      dbStatus = false;
+    } else {
+      try {
+        Pong pong = influxDB.ping();
+        String version = pong.getVersion();
+        // may be check for version >= 0.9
+        if (version != null && !version.contains("unknown")) {
+          dbStatus = true;
+          logger.debug("database status is OK");
+        } else {
+          logger.error("database connection failed, version is: \"{}\" response time was \"{}\"",
+              version, pong.getResponseTime());
+          dbStatus = false;
+        }
+      } catch (RuntimeException e) {
+        dbStatus = false;
+        logger.error("database connection failed throwing an exception");
+        handleDatabaseException(e);
+      }
+    }
+    return dbStatus;
+  }
+
+  private void disconnect() {
+    influxDB = null;
+    connected = false;
+  }
+
+  private boolean isConnected() {
+    return connected;
+  }
+
+  @Override
+  public String getName() {
+    return "influxdb";
+  }
+
+  @Override
+  public void store(Item item) {
+    store(item, null);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void store(Item item, String alias) {
+    if (item.getState() instanceof UnDefType) {
+      return;
+    }
+
+    if (!isProperlyConfigured) {
+      logger.warn("Configuration for influxdb not yet loaded or broken.");
+      return;
+    }
+
+    if (!isConnected()) {
+      logger.warn("InfluxDB is not yet connected");
+      return;
+    }
+
+    String realName = item.getName();
+    String name = (alias != null) ? alias : realName;
+
+    State state = null;
+    if (item.getAcceptedCommandTypes().contains(HSBType.class)) {
+      state = item.getStateAs(HSBType.class);
+      logger.trace("got hsbtype", state.toString());
+    } else if (item.getAcceptedDataTypes().contains(PercentType.class)) {
+      state = item.getStateAs(PercentType.class);
+      logger.trace("percentype item found {}", name);
+    } else {
+      logger.trace("other item found");
```

Comment:
better unify the different states to be logged here in order to ease debugging. Something linke "Tried to get item as {}"